source/documentation/parameters.md

```diff
@@ -0,0 +1,160 @@
+
+The users who have access to a repo does change frequently. It's quite normal to give more people access to a repo than are actually working on the code - maybe it's one repo of many that a team is given access to. Or maybe everyone in the org or GitHub Enterprise Account have access. Someone in the future might give another team access to a private repo, or flip it to internal or public access. It's easy to forget that a repo has a secret that should not be shared.
+
+In conclusion, you can easily lose control of secrets if you ever put them in a git repository, because they hide well, can stick around in the history for years, and access control can change over time, due to different motivations for sharing code compared to secrets.
+
+### What to do in the event of a secret being committed to a git repo
+
+* Revoke the secret (if you can)
+* [Report security concerns](information-governance.html#reporting-security-incidents)
+* Delete relevant files from the git history using a tool like [bfg](https://rtyley.github.io/bfg-repo-cleaner/). Then force push the cleaned repo to Github. Also search for forks and copies of your repo.
+
+## A file which is not committed to git
+
+You can store a secret in a file in your Analytical Platform home directory, just don't add it to a git repo. When you work in RStudio or Jupyter and start off exploring an API or other external service, this can often be fine. You could read it in like this:
+
+```python
+    file  = open('../secrets/apikey.txt', 'r')
+    api_key = file.read()
+```
+
+The key risk that you need to manage is the chance of accidentally adding the file with the secret to your git repo. You could add the filename to your .gitignore file, but it's probably more reliable if you put it in a directory outside any repo.
+
+This technique is limited to personal use. Whilst you *could* share the file securely with colleagues, you do lose control and oversight of what they do with it. And it is not suitable for apps or Airflow pipelines. For these needs, store the secret in the cloud using parameters or an S3 bucket
+
+## S3 bucket
+
+You can put a secret in a file in an S3 bucket. This allows you to control access from individual platform users, apps and Airflow.
+
+However this may not be as quick to access or as convenient as using the Parameter Store. Also you may need different permissions for the secret file compared to your data, which means creating separate buckets.
+
+### Example - Airflow gets secret from S3
+
+1. Create 'secrets.json' with these contents:
```

Comment:
I think this could do with more introduction and to set the scene that this is a generic example (that users would not be able to run exactly themselves).